python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -93,41 +120,73 @@ cdef class Scalar:
                 f"{type(value).__name__} to cudf scalar"
             )
 
+    def get_device_value(self):
```

Comment:
Should we make this an internal function? I imagine we don't want someone calling it directly ever. We want them to go through the `value` property.

Reply:
Done.